### ISSUE TITLE
@everywhere not update function definition

### ISSUE THREAD

**Issue by phlavenk** [JuliaLang/julia#20208]
It seems to me that the @everywhere macro has problems with function redefinition on the julia master process (the REPL julia) on 0.5.0
Short demonstration:
```julia
julia> addprocs(2);

julia> @everywhere function test(i)
           println("Version 1 : $i")
       end

julia> pmap(test, 1:2);
        From worker 2:  Version 1 : 1
        From worker 3:  Version 1 : 2

julia> map(test, 1:2);
Version 1 : 1
Version 1 : 2
```
Untilla now, everything works fine. But I'd like to redefine the function test:
```julia
julia> @everywhere function test(i)
           println("Version 2 : $i")
       end
WARNING: Method definition WARNING: Method definition testtest((AnyAny)) in modu
le Main in module Main at REPL[2]:2 at REPL[2]:2 overwritten overwritten at REPL
[5]:2 at REPL[5]:2.
.
WARNING: Method definition test(Any) in module Main at REPL[2]:2 overwritten at
REPL[5]:2.

julia> pmap(test, 1:2);
        From worker 3:  Version 2 : 2
        From worker 2:  Version 2 : 1

julia> map(test, 1:2);
Version 1 : 1
Version 1 : 2
```
For some reason, the redefinition of the function test in @everywhere macro didn't happen on the master julia, although has been aplied correctly on the worker julia processes.

```julia
julia> versioninfo()
Julia Version 0.5.0
Commit 3c9d753 (2016-09-19 18:14 UTC)
Platform Info:
  System: NT (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i5-4200M CPU @ 2.50GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, haswell)
```

**Comment by amitmurthy** [comment]
I think it is #265. This has been fixed on master.  

On 0.5
```
julia> test(1)
Version 2 : 1

julia> map(test, 1:2);
Version 1 : 1
Version 1 : 2
```

Maybe @vtjnash / others can suggest a workaround for 0.5 ?

**Comment by phlavenk** [comment]
Thank you Amit,
this explains the behavior. Looking forward to 0.6. Untill then, one has to be aware of it and rely on restarting julia kernel more often during interactive development.

Please @amitmurthy , close the issue as solved for 0.6.
Petr
